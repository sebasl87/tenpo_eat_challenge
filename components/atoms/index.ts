export { default as RestaurantsLayout } from './RestaurantsLayout';
