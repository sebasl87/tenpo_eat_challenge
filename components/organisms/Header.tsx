import React from 'react'
import { Image, Text, View } from 'react-native';
import { styles } from '../../styles/styles';

function Header() {
    return (
        <>
            <View style={styles.header}>
                <Image
                    source={require("../../assets/imgs/user.png")}
                    style={styles.userImg}
                />
                <Image
                    source={require("../../assets/icons/search.png")}
                    style={styles.searchImg}
                />
            </View>
            <View style={styles.header}>
                <View style={{ flexDirection: "column" }}>
                    <Image
                        source={require("../../assets/imgs/logo.png")}
                        style={styles.logoImg}
                    />
                    <Text>DELIVER APP</Text>
                </View>
                <View style={{ flexDirection: "row", marginBottom: -16 }}>
                    <Image
                        source={require("../../assets/imgs/hand.png")}
                        style={styles.handImg}
                    />
                    <View style={{ top: "100%", height: 6 }}>

                        <Image
                            source={require("../../assets/icons/ellipse.png")}
                            style={styles.ellipseImg}
                        />
                    </View>
                </View>
            </View>
        </>
    );
}

export default Header