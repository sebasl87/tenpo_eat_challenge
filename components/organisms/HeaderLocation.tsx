import React from 'react'
import { Image, Text, TextInput, View } from 'react-native';
import { styles } from '../../styles/styles';

function Header() {
    return (
        <>
            <View style={styles.headerLocation}>
                <View style={{ marginBottom: 32, display: "flex", flexDirection: "row" }}>
                    <Image
                        source={require("../../assets/icons/location.png")}
                        style={{ height: 24, width: 24, marginRight: 6, paddingTop: 30 }}
                    />
                    <Text style={styles.fontTitle}>Agregar dirección de la entrega</Text>
                </View>
            </View>
            <TextInput placeholder='Escribe tu dirección' style={styles.inputLocation} />
        </>
    );
}

export default Header