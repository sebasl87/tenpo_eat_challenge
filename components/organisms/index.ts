export { default as Header } from "./Header";
export { default as HeaderLocation } from "./HeaderLocation";
