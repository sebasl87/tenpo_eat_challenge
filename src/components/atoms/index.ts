import AddressLayout from './AddressLayout';
export { default as RestaurantsLayout } from './RestaurantsLayout';
export { default as CategoriesLayout } from './CategoriesLayout';
export { default as HeaderFavorites } from './HeaderFavorites';
export { default as AddressLayout } from './AddressLayout';
