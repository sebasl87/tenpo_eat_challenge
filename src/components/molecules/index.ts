export { default as DetailCard } from './DetailCard';
export { default as DiscountCircle } from './DiscountCircle';
export { default as LogoRounded } from './LogoRounded';
export { default as DetailFavorites } from './DetailFavorites';
