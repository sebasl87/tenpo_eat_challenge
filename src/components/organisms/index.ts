export { default as Header } from "./Header";
export { default as HeaderLocation } from "./HeaderLocation";
export { default as ContainerMapView } from "./ContainerMapView";
export { default as RestaurantRow } from "./RestaurantRow";
export { default as ContainerAddressData } from "./ContainerAddressData";
export { default as CategoriesRow } from "./CategoriesRow";
export { default as FavoritesRow } from "./FavoritesRow";
export { default as HomeLocation } from './HomeLocation';
