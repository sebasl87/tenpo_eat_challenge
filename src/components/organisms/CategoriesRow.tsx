import React from 'react'
import { View, ScrollView, Text } from 'react-native';
import styled from 'styled-components/native';
import { CategoriesLayout } from '../atoms'

export interface ICategories { }

const Title = styled.Text`
  color: #333333;
  font-size: 18px;
  margin-bottom: 6px;
  margin-top: 32px;
`;

const linksData = [
    {
        uri: require("../../../assets/imgs/hamburguer.png"),
        category: "HAMBURGUESAS"
    },
    {
        uri: require("../../../assets/imgs/italian.png"),
        category: "ITALIANA"
    },
    {
        uri: require("../../../assets/imgs/pizza.png"),
        category: "PIZZAS"
    },

]

const CategoriesRow: React.FC<ICategories> = () => (
    <>

        <Title style={{ fontFamily: "Gotham-Bold" }}>CATEGORIAS</Title>
        <ScrollView horizontal={true}>
            <View style={{ flexDirection: "row" }}>
                {linksData.map((item, i) => (
                    <View key={i} style={{ marginRight: 6 }}>

                        <CategoriesLayout image={item.uri}>{item.category}</CategoriesLayout>
                    </View>
                ))}
            </View >
        </ScrollView>

    </>
);


export default CategoriesRow;