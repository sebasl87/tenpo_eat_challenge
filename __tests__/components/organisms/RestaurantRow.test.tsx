import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import { RestaurantRow } from "../../../src/components/organisms";

export const mock_restoList = [{
    id: 1,
    uri: "https://i.imgur.com/1PLcyBL.png",
    uri_large: "https://i.imgur.com/25zvKsX.png",
    discount: "50",
    restaurantName: "Mcdonalds",
    rate: "3.5",
    lapseTime: "10-50",
    bgColor: ""
}, {
    id: 2,
    uri: "https://i.imgur.com/PPaKrFj.png",
    uri_large: "https://i.imgur.com/Ui5IcDP.png",
    discount: "30",
    restaurantName: "MELT pizzas",
    rate: "4.5",
    lapseTime: "10-60",
    bgColor: ""
}]

describe("RestaurantRow component", () => {
    it("should render successfully with RestaurantRow_mock", () => {
        const restaurantRowSection = renderer
            .create(<RestaurantRow restoList={mock_restoList} />)
            .toJSON();

        expect(restaurantRowSection).toMatchSnapshot();
    });
});
