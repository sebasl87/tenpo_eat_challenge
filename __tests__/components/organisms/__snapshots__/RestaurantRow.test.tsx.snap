// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantRow component should render successfully with RestaurantRow_mock 1`] = `
[
  <Text
    style={
      [
        {
          "color": "#333333",
          "fontSize": 18,
          "marginBottom": 6,
        },
        {
          "fontFamily": "Gotham-Bold",
        },
      ]
    }
  >
    RESTAURANTES
  </Text>,
  <RCTScrollView
    horizontal={true}
  >
    <View>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="clickMeRestaurantRow"
        >
          <View
            style={
              {
                "marginRight": 12,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#00BAA4",
                    "height": 30,
                    "justifyContent": "center",
                    "marginLeft": 78,
                    "marginTop": 2,
                    "position": "absolute",
                    "width": 30,
                    "zIndex": 2,
                  },
                  {
                    "borderRadius": 50,
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#fff",
                      "fontSize": 11,
                    },
                    {
                      "fontFamily": "Roboto-Light",
                    },
                  ]
                }
              >
                50
                %
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#fff",
                      "fontSize": 7,
                      "marginTop": -2,
                    },
                    {
                      "fontFamily": "Roboto-Light",
                    },
                  ]
                }
              >
                DCTO
              </Text>
            </View>
            <View
              bgColor=""
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "",
                    "borderRadius": 16,
                    "display": "flex",
                    "height": 104,
                    "justifyContent": "center",
                    "marginTop": 10,
                    "width": 104,
                  },
                ]
              }
            >
              <Image
                source={
                  {
                    "uri": "https://i.imgur.com/1PLcyBL.png",
                  }
                }
                style={
                  [
                    {
                      "height": "100%",
                      "maxWidth": 104,
                      "width": "100%",
                    },
                  ]
                }
              />
            </View>
            <View>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "display": "flex",
                      "maxWidth": 104,
                      "width": "100%",
                    },
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#333333",
                        "fontSize": 14,
                      },
                      {
                        "fontFamily": "Gotham-Medium",
                      },
                    ]
                  }
                >
                  Mcdonalds
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "maxWidth": 104,
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "display": "flex",
                        "flexDirection": "row",
                      },
                    ]
                  }
                >
                  <Image
                    source={
                      {
                        "testUri": "../../../assets/imgs/starmini.png",
                      }
                    }
                    style={
                      [
                        {
                          "height": 16,
                          "marginRight": 2,
                          "width": 16,
                        },
                      ]
                    }
                  />
                  <Text
                    style={
                      [
                        {
                          "color": "#333333",
                          "fontSize": 12,
                        },
                        {
                          "fontFamily": "Gotham-Medium",
                        },
                      ]
                    }
                  >
                    3.5
                  </Text>
                </View>
                <View>
                  <Text
                    style={
                      [
                        {
                          "color": "#333333",
                          "fontSize": 12,
                        },
                        {
                          "fontFamily": "Gotham-Medium",
                        },
                      ]
                    }
                  >
                    10-50
                     min
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="clickMeRestaurantRow"
        >
          <View
            style={
              {
                "marginRight": 12,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#00BAA4",
                    "height": 30,
                    "justifyContent": "center",
                    "marginLeft": 78,
                    "marginTop": 2,
                    "position": "absolute",
                    "width": 30,
                    "zIndex": 2,
                  },
                  {
                    "borderRadius": 50,
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#fff",
                      "fontSize": 11,
                    },
                    {
                      "fontFamily": "Roboto-Light",
                    },
                  ]
                }
              >
                30
                %
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#fff",
                      "fontSize": 7,
                      "marginTop": -2,
                    },
                    {
                      "fontFamily": "Roboto-Light",
                    },
                  ]
                }
              >
                DCTO
              </Text>
            </View>
            <View
              bgColor=""
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "",
                    "borderRadius": 16,
                    "display": "flex",
                    "height": 104,
                    "justifyContent": "center",
                    "marginTop": 10,
                    "width": 104,
                  },
                ]
              }
            >
              <Image
                source={
                  {
                    "uri": "https://i.imgur.com/PPaKrFj.png",
                  }
                }
                style={
                  [
                    {
                      "height": "100%",
                      "maxWidth": 104,
                      "width": "100%",
                    },
                  ]
                }
              />
            </View>
            <View>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "display": "flex",
                      "maxWidth": 104,
                      "width": "100%",
                    },
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#333333",
                        "fontSize": 14,
                      },
                      {
                        "fontFamily": "Gotham-Medium",
                      },
                    ]
                  }
                >
                  MELT pizzas
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "maxWidth": 104,
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "display": "flex",
                        "flexDirection": "row",
                      },
                    ]
                  }
                >
                  <Image
                    source={
                      {
                        "testUri": "../../../assets/imgs/starmini.png",
                      }
                    }
                    style={
                      [
                        {
                          "height": 16,
                          "marginRight": 2,
                          "width": 16,
                        },
                      ]
                    }
                  />
                  <Text
                    style={
                      [
                        {
                          "color": "#333333",
                          "fontSize": 12,
                        },
                        {
                          "fontFamily": "Gotham-Medium",
                        },
                      ]
                    }
                  >
                    4.5
                  </Text>
                </View>
                <View>
                  <Text
                    style={
                      [
                        {
                          "color": "#333333",
                          "fontSize": 12,
                        },
                        {
                          "fontFamily": "Gotham-Medium",
                        },
                      ]
                    }
                  >
                    10-60
                     min
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;
